src/gradle.ts

```diff
@@ -106,6 +101,21 @@ export function getTaskToPublish(
             "publishToSonatype",
             "closeAndReleaseSonatypeStagingRepository",
           ];
+        } else if (
+          line.startsWith("publishAllPublicationsToMavenCentralRepository")
+        ) {
+          if (tasks.length !== 0 && tasks[0] === "publish") {
```

Comment:
This is necessary because the tasks generated with the plugin applied look like this

```plaintext
Publishing tasks
----------------
generateMetadataFileForMavenPublication - Generates the Gradle metadata file for publication 'maven'.
generatePomFileForMavenPublication - Generates the Maven POM file for publication 'maven'.
publish - Publishes all publications produced by this project.
publishAllPublicationsToMavenCentralRepository - Publishes all Maven publications produced by this project to the mavenCentral repository.
publishMavenPublicationToMavenCentralRepository - Publishes Maven publication 'maven' to Maven repository 'mavenCentral'.
publishMavenPublicationToMavenLocal - Publishes Maven publication 'maven' to the local Maven repository.
publishToMavenLocal - Publishes all Maven publications produced by this project to the local Maven cache.
```

but we want to ignore the catch-all `publish` task and instead use the more specific `publishMavenPublicationToMavenCentralRepository` task instead.